src/Rule/RedirectedLink.php

```diff
@@ -33,35 +32,28 @@ public function check()
 		return count($this->issues);
 	}
 
-	private function checkLink($links) {
-		$curls = array();
-		$mcurl = curl_multi_init();
-		foreach (array_keys($links) as $i => $link) {
-			$curls[$i] = curl_init();
-			curl_setopt($curls[$i], CURLOPT_URL, $link);
-			curl_setopt($curls[$i], CURLOPT_HEADER, true);
-			curl_setopt($curls[$i], CURLOPT_NOBODY, true);
-			curl_setopt($curls[$i], CURLOPT_REFERER, true);
-			curl_setopt($curls[$i], CURLOPT_TIMEOUT, 2);
-			curl_setopt($curls[$i], CURLOPT_TIMEOUT, 2);
-			curl_setopt($curls[$i], CURLOPT_AUTOREFERER, true);
-			curl_setopt($curls[$i], CURLOPT_RETURNTRANSFER, true);
-			curl_setopt($curls[$i], CURLOPT_FOLLOWLOCATION, true);
-			curl_multi_add_handle($mcurl, $curls[$i]);
-		}
+	private function checkLink($element, $link) {
+		$curl = curl_init();
+		
+		curl_setopt($curl, CURLOPT_URL, $link);
+		curl_setopt($curl, CURLOPT_HEADER, true);
+		curl_setopt($curl, CURLOPT_NOBODY, true);
+		curl_setopt($curl, CURLOPT_REFERER, true);
+		curl_setopt($curl, CURLOPT_TIMEOUT, 2);
+		curl_setopt($curl, CURLOPT_AUTOREFERER, true);
+		curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+		curl_setopt($curl, CURLOPT_FOLLOWLOCATION, true);
+		
 		$running = null;
 		do {
-			curl_multi_exec($mcurl, $running);
+			curl_exec($curl);
 		} while ($running > 0);
-		foreach (array_keys($links) as $i => $link) {
-			$status = curl_getinfo($curls[$i], CURLINFO_RESPONSE_CODE);
+			$status = curl_getinfo($curl, CURLINFO_RESPONSE_CODE);
 			// If the status is 400 or greater the link is broken so dont bother checking.
 			if ($status < 400) {
-				$this->checkRedirect($links[$link]);
+				$this->checkRedirect($link);
```

Comment:
This should be `if ($status < 400) {
				$this->checkRedirect($element);
			}`

The phpunit tests are failing because we call getAttribute on a string by passing in the link.

Reply:
Or alternatively we pass in the link and don't call getAttribute on it since we already have it.